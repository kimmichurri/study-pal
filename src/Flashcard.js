import React, { Component } from 'react';
import DisplayQuestionsAndAnswers from './DisplayQuestionsAndAnswers';

export default class Flashcard extends Component {
    constructor() {
        super();
        this.state = {
            correctAnswer: null,
            currentFlashcardIndex: 0
        }
    }

    validateAnswer =(e) => {
        if (e.target.innerText === this.props.flashcards[0].correctAnswer) {
            console.log("You're Right!")
            this.setState({
                correctAnswer: true
            })
        } else {
            console.log("Sorry, that is incorrect!")
            this.setState({
                correctAnswer: false
            })
        }
    }

    showNextFlashcard = (e) => {
        this.setState({
            correctAnswer: null,
            currentFlashcardIndex: this.state.currentFlashcardIndex + 1
        })
        if (e.target.innerText === "Keep Trying!") {
            console.log("put this into study bank")
        } 
    }
    

    render() {
        return (
            <article className="flashcard-container">
                <DisplayQuestionsAndAnswers 
                    flashcard={this.props.flashcards[this.state.currentFlashcardIndex]} 
                    validateAnswer={this.validateAnswer}
                    correctAnswer={this.state.correctAnswer}
                    showNextFlashcard={this.showNextFlashcard}
                    />
            </article>
        )
    }
    
}