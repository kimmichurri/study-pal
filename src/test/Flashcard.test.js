import React from 'react';
import Flashcard from '../Flashcard';
import { shallow } from 'enzyme';

const mockFlashcards = [
    {
        "id": "a257e858-36c6-11e9-b210-d663bd873d93",
        "category": "String Methods",
        "question": "Which method allows you to remove whitespace from either end of a string?",
        "answerChoices": [".trim()", ".slim()", ".clip()"],
        "correctAnswer": ".trim()",
        "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim"
    }
]

let mockCorrectAnswer;

describe('Flashcard', () => {
    
    let wrapper;

    it('should have the proper default state', () => {

        const wrapper = shallow(
            <Flashcard 
                flashcards={mockFlashcards}
            />
        );
        expect(wrapper.state()).toEqual({
            correctAnswer: null
        });
    });

    it.skip('should change the state of correct answer to true if correct answer is selected', () => {

        const wrapper = shallow(
            <Flashcard 
                flashcards={mockFlashcards}
            />
        );
        mockCorrectAnswer = true;
        wrapper.instance().validateAnswer();
        wrapper.instance().state.correctAnswer = mockCorrectAnswer;
        expect(wrapper.state()).toEqual({
            correctAnswer: true
        });
    });

    it('should change the state of correct answer to false if correct answer is not selected', () => {
        
        const wrapper = shallow(
            <Flashcard 
                flashcards={mockFlashcards}
            />
    });

    it('should push an incorrectly answered flashcard into the study bank', () => {

        const wrapper = shallow(
            <Flashcard 
                flashcards={mockFlashcards}
            />
    });

    it('should reset correct answer to null and call incrementCurrentIndex function', () => {

        const wrapper = shallow(
            <Flashcard 
                flashcards={mockFlashcards}
            />
    });

    it('should match the snapshot when rendering', () => {

        const wrapper = shallow(
            <Flashcard 
                flashcards={mockFlashcards}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

});