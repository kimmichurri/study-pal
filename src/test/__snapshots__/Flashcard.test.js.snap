// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flashcard should match the snapshot when rendering 1`] = `
<article
  className="flashcard-container"
>
  <DisplayQuestionsAndAnswers
    allFlashcards={
      Array [
        Object {
          "answerChoices": Array [
            ".trim()",
            ".slim()",
            ".clip()",
          ],
          "category": "String Methods",
          "correctAnswer": ".trim()",
          "id": "a257e858-36c6-11e9-b210-d663bd873d93",
          "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim",
          "question": "Which method allows you to remove whitespace from either end of a string?",
        },
      ]
    }
    correctAnswer={null}
    showNextFlashcard={[Function]}
    validateAnswer={[Function]}
  />
</article>
`;
